web/frontend/src/components/gantry/index.svelte

```diff
@@ -1,132 +1,253 @@
 <script lang="ts">
 
-import { grpc } from '@improbable-eng/grpc-web';
-import { Client, gantryApi } from '@viamrobotics/sdk';
-import { displayError } from '@/lib/error';
-import { rcLogConditionally } from '@/lib/log';
-import Collapse from '@/components/collapse.svelte';
-
-export let name: string;
-export let client: Client;
-export let status: {
-  is_moving: boolean
-  lengths_mm: number[]
-  positions_mm: number[]
-} = {
-  is_moving: false,
-  lengths_mm: [],
-  positions_mm: [],
-};
-
-$: parts = status.lengths_mm.map((_, index) => ({
-  axis: index,
-  pos: status.positions_mm[index]!,
-  length: status.lengths_mm[index]!,
-}));
-
-$: if (status.lengths_mm.length !== status.positions_mm.length) {
-  console.error('gantry lists different lengths');
-}
-
-const increment = (axis: number, amount: number) => {
-  if (!status) {
-    return;
+  import { grpc } from '@improbable-eng/grpc-web';
+  import { Client, gantryApi } from '@viamrobotics/sdk';
+  import type { ServiceError } from '@viamrobotics/sdk';
+  import { displayError } from '@/lib/error';
+  import { rcLogConditionally } from '@/lib/log';
+  import Collapse from '@/components/collapse.svelte';
+  
+  export let name: string;
+  export let client: Client;
+  export let status: {
+    is_moving: boolean
+    lengths_mm: number[]
+    positions_mm: number[]
+  } = {
+    is_moving: false,
+    lengths_mm: [],
+    positions_mm: [],
+  };
+  
+  interface GantryStatus {
+    pieces: {
+      axis: number,
+      pos: number,
+      length: number,
+    }[]
   }
+  
+  let modifyAllStatus: GantryStatus = {
+    pieces: [],
+  };
+  
+  const gantryModifyAllDoMoveToPosition = async () => {
+    const gantry = status!;
+    let newList = new Array<number>(gantry.positions_mm.values.length); 
+    const newPieces = modifyAllStatus.pieces;
 
-  const pos: number[] = [];
-
-  for (const [i, part] of parts.entries()) {
-    pos[i] = part!.pos;
+    for (let i = 0; i < newPieces.length; i += 1) {
+      newList[i] = newPieces[i]!.pos
+    }
+  
+    try {
+      let req = new gantryApi.MoveToPositionRequest();
+      req.setName(name);
+      req.setPositionsMmList(newList);
+    
+      rcLogConditionally(req);
+      client.gantryService.moveToPosition(req, new grpc.Metadata(), displayError);
+    } catch (error) {
+      displayError(error as ServiceError);
+    }
+  
+    modifyAll = false;
+  };
+  
+  const gantryHome = async () => {
+    try {
+      let req = new gantryApi.HomeRequest();
+      req.setName(name);
+    
+      rcLogConditionally(req);
+      client.gantryService.home(req, new grpc.Metadata(), displayError);
```

Comment:
I can't figure out why, but this line causes runtime error: invalid memory address or nil pointer dereference. Not sure why this would be happening for here, but not with moveToPosition even though they are set up the same, so I'm wondering if anyone has seen something like this before and/or has suggestions?